base/fs.h

```diff
@@ -65,7 +65,20 @@ namespace base {
   // elements (is a normalized path).
   std::string get_absolute_path(const std::string& path);
 
-  paths list_files(const std::string& path);
+  // Item types that list_files() can be filtered by
+  enum class ItemType {
+      All,
+      Directories,
+      Files
+  };
```

Comment:
I think I'll keep a simple enum class since you can't really mix and match flags in this case, if you want Directories | Files you use "All", and if you want Directories | All... you... well you shouldn't want that, that makes no sense 🤪